ext/libdatadog_api/feature_flags.c

```
@@ -0,0 +1,548 @@
+
+  const ddog_ffe_Handle_Configuration config =
+    (ddog_ffe_Handle_Configuration)rb_check_typeddata(self, &configuration_data_type);
+  const ddog_ffe_ExpectedFlagType expected_ty = expected_type_from_value(expected_type);
+  ddog_ffe_Handle_EvaluationContext context = evaluation_context_from_hash(context_hash);
+
+  ddog_ffe_Handle_ResolutionDetails resolution_details = ddog_ffe_get_assignment(
+    config,
+    RSTRING_PTR(flag_key),
+    expected_ty,
+    context
+  );
+
+  ddog_ffe_evaluation_context_drop(&context);
+
+  return TypedData_Wrap_Struct(resolution_details_class, &resolution_details_typed_data, resolution_details);
+}
+
+static void resolution_details_free(void *ptr) {
+  ddog_ffe_Handle_ResolutionDetails resolution_details = (ddog_ffe_Handle_ResolutionDetails)ptr;
+  ddog_ffe_assignment_drop(&resolution_details);
+}
+
+/*
+ * call-seq:
+ *   resolution_details.raw_value() -> Object
+ *
+ * Get the raw resolved value from libdatadog.
+ *
+ * The value can be any type depending on the feature flag (String, Integer, Float, Boolean, or nil).
+ * For object types, returns the raw JSON string without parsing.
+ */
+static VALUE resolution_details_get_raw_value(VALUE self) {
+  ddog_ffe_Handle_ResolutionDetails resolution_details =
+    (ddog_ffe_Handle_ResolutionDetails)rb_check_typeddata(self, &resolution_details_typed_data);
+
+  struct ddog_ffe_VariantValue value = ddog_ffe_assignment_get_value(resolution_details);
+
+  switch (value.tag) {
+    case DDOG_FFE_VARIANT_VALUE_STRING:
+      return str_from_borrow(value.string);
+    case DDOG_FFE_VARIANT_VALUE_INTEGER:
+      return LONG2NUM(value.integer);
+    case DDOG_FFE_VARIANT_VALUE_FLOAT:
+      return rb_float_new(value.float_);
+    case DDOG_FFE_VARIANT_VALUE_BOOLEAN:
+      return value.boolean ? Qtrue : Qfalse;
+    case DDOG_FFE_VARIANT_VALUE_OBJECT:
+      return str_from_borrow(value.string);
+    case DDOG_FFE_VARIANT_VALUE_NONE:
+    default:
+      return Qnil;
```

Should the default case raise an exception instead of being treated as None?